### PR TITLE
Writing for section 15

### PR DESCRIPTION
**Pull request by karanbayns**

### CODE REVIEW

File: book/15-ordinal-logistic.qmd [hunk 1]
```diff
@@ -76,3 +76,2760 @@ mindmap
...
+
+::: {.LO}
+::::{.LO-header}
+Learning Objectives
+::::
+::::{.LO-container}
+By the end of this chapter, you will be able to:
+
+- Describe why ordinary **linear models** and **multinomial logistic regression** are inappropriate for **ordered categorical outcomes**.
+
```
Comment (pnickchi) [inline book/15-ordinal-logistic.qmd:178]: Add an item for Ordinal variable as well.

File: book/15-ordinal-logistic.qmd [hunk 2]
```diff
@@ -76,3 +76,2760 @@ mindmap
...
+The dataset contains the following variables:
+
+- `likelihood_to_apply`: how likely the student will apply to graduate school, **a discrete and ordinal response with four increasing categories** (`Low`, `Medium`, `High` and `Very High`).
+- `GPA`: the student's current GPA, **a continuous explanatory variable**.
+- `GRE_score`: the student's most recent GRE score, **a continuous explanatory variable**.
+- `field_of_study`: the student's undergraduate field of study, **a categorical explanatory variable with four categories** (Arts, Business, Social Sciences, and STEM).
+
+::: {.panel-tabset}
+
+## R Code
```
Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: For all the R code and Python code parts, please use a smaller header size so that they are not the main section. For example you can add ### R Code or #### R Code. Let's keep them inside the sections. Also have a look at [this ](https://alexrod61.github.io/regression-cookbook/book/01-intro.html#example-eda-for-housing-data)for a guidline on how to write the Python and R code tabset. We want to have one tabset for code and one for the result of code.

File: book/15-ordinal-logistic.qmd [hunk 3]
```diff
@@ -76,3 +76,2760 @@ mindmap
...
+- How are a student's GPA, GRE score, and field of study **associated** with their likelihood of applying to graduate school?
+- Can we accurately **predict** a student's likelihood of applying to graduate school based on their GPA, GRE score, and field of study?
+
+These questions correspond to two distinct **statistical objectives**:
+- **Inference**: Estimate the association between explanatory variables (GPA, GRE score, and field of study) and the likelihood of applying to graduate school.
+- **Prediction**: Develop a classification model that classifies individuals into their most probable ordinal category for the likelihood of applying to graduate school: `Low`, `Medium`, `High`, or `Very High`.
+
+## Data Collection and Wrangling
+
+The dataset contains the following variables:
```
Comment (pnickchi) [inline book/15-ordinal-logistic.qmd:259]: Have a look at [this ](https://alexrod61.github.io/regression-cookbook/book/01-intro.html#example-eda-for-housing-data)guidline for how to define the variables and their description in a table

File: book/15-ordinal-logistic.qmd [hunk 4]
```diff
@@ -76,3 +76,2760 @@ mindmap
...
+
+$$
+\text{logit}(P(Y \le j \mid \underline{X})) = \alpha_j - (\beta_1 X_1 + \beta_2 X_2 + \ldots + \beta_K X_K), \quad j = 1, \ldots, J-1,
+$$
+
+where $\alpha_1 < \alpha_2 < \cdots < \alpha_{J-1}$ are the category-specific intercepts (also called **thresholds** or **cutpoints**) and $\beta_1, \ldots, \beta_K$ are the shared slope coefficients. Note the sign convention: the predictor terms are subtracted from the intercept so that a positive $\beta_k$ corresponds to higher values of $X_k$ shifting probability mass toward higher categories. This sign convention is standard in R's `MASS::polr` and in most textbooks, and it means $\alpha_j$ increases with $j$ to preserve the ordering of cumulative probabilities.
+
+The total number of parameters to estimate is $(J - 1) + K$: one threshold per cumulative split, plus one slope per predictor.
+
+### Category Probabilities from the Cumulative Model
```
Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Suggest to change to: From Cumulative Odds to Category Probabilities

File: book/15-ordinal-logistic.qmd [hunk 5]
```diff
@@ -76,3 +76,2760 @@ mindmap
...
+
+where $f(\cdot) = F(\cdot)(1 - F(\cdot))$ is the logistic density. The partial derivative with respect to each shared slope $\beta_k$ is:
+
+$$
+\frac{\partial \ell}{\partial \beta_k} = -\sum_{i=1}^n X_{i,k} \sum_{j=1}^{J} y_{i,j} \left[ \frac{f(\alpha_j - \boldsymbol{\beta}^\top \underline{X}_i)}{p_j(\underline{X}_i)} - \frac{f(\alpha_{j-1} - \boldsymbol{\beta}^\top \underline{X}_i)}{p_j(\underline{X}_i)} \right].
+$$
+
+The key difference from the multinomial case is that the slopes $\beta_k$ appear in **all** $J-1$ cumulative logits simultaneously, so their score components aggregate information across every cumulative split.
+
+**Hessian matrix.** The Hessian $H(\boldsymbol{\theta})$ consists of second-order partial derivatives of $\ell$. Because the shared slopes couple all cumulative logits together, the Hessian is a dense $(J-1+K) \times (J-1+K)$ matrix — unlike the block-diagonal structure seen in multinomial logistic regression. In software this is handled automatically.
```
Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: This is great to go over the details. I think you can have them in a additional note box. Take a look at [this ](https://alexrod61.github.io/regression-cookbook/book/01-intro.html#let-the-cooking-begin) guidline and see how a tip box is defined. Please include a reference here too.

**Comment by pnickchi** [comment]
@karanbayns Thanks Karan for working on this Chapter. I think generally it looks good. I had a couple of comments to make the chapter more consistent with other chapters that we already have. You can see my comments in qmd file under your last commits. Also some minor comments about the content. Please apply them and render the book. Once it is done let me know and I will pull the changes from this branch and give you the rest of my comments which are more structural. Thanks!

File: book/15-ordinal-logistic.qmd
Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: @karanbayns Add the ordinal levels for all the examples (customer satisfaction, educational, ect)

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Please remove $x$s - too soon to define notations here

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: The cross references here does not render on html file. Can you please check?

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Let's frame this as: understanding the relationship between regressors and the inclination to apply to graduate school

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Remove the dots at the end

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd:836]: Looks much better, thanks for making the changes. Just a final polish to change the dark text and purple background

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd:1040]: The x/y label font size is small in python version. Can we make the font size bigger?

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Add a cross reference to binary logistic section as a reminder for reader

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: I think it is better to use the lower case k for the number of regressors. Please change it across the document. This applies to \beta_k instead of \beta_K

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Here right after "be the cumulative probability." let's add a short paragraph where we explain how the formula looks like for some example values of J. Start with J=3 and show how the P(Y_i \leq j | x) looks like for this J and then tell the reader the same idea applies to other values of J. I think this helps them see the formula and connection easier. Then continue with what you have already

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: I suggest these changes: "The proportional odds model assumes that each regressor has the same effect across all cumulative logits. In other words, the regression coefficient for a given regressor is shared across the cumulative comparisons and does not depend on the response category cutoff (i.e. the value of $j$). This assumption leads to a simpler model with a simple interpretation: each coefficient describes how the regressor shifts the odds of being in a higher response category, regardless of where the ordinal response is split. In contrast, the non-proportional odds model relaxes this assumption by allowing regression coefficients to vary across cumulative splits of the response. In the following sections, we introduce these two approaches in detail and discuss their interpretation and practical implications. Later in the chapter, we also show how to formally assess the proportional odds assumption using the Brant–Wald test."

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: I think this formula is missing a log() in the right hand side?

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Let's add a bit of more math details here so that it is clear where this formula is coming from. Start by writing the logit( P(Y<j | x) ) and show how we derive the systematic component from it. Then show how we replace it to get the formula you have now. Happy to discuss in our meeting if that helps

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Suggest to change the first part to: We first encode each individual’s observed response category using indicator variables. Let $y_{i,j} = 1$ if the $i$-th individual belongs to category $j$, and let $y_{i,j} = 0$ otherwise. Since each individual belongs to exactly one of the (J) possible categories, the vector $(y_{i,1}, y_{i,2}, \ldots, y_{i,J})$ contains exactly one entry equal to (1), with all remaining entries equal to (0). This representation is often called a one-hot encoding of the response category.

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: I think this should be "every term equals $0$ except the one matching the individual's true category,"

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Please check [here ](https://alexrod61.github.io/regression-cookbook/book/02-stats-review.html#sec-mle-regression-engine)for the likelihood notation in this book.

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Mention that we describe it in the tip box as additional details. Does the algorithm have a specific name, like a flavor of Newthon-Raphson? Adding a reference for it would be nice

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: I wonder if it makes more sense to move the header (Firring the Proportional Odds Model) right after the Estimation header? The rationale is you already start the section with talking about the proportional odds model.

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: I think "Note: polr() uses the sign convention logit P(Y <= j) = alpha_j - beta * X" can be a source of confusion for reader. If polr() goes with the sign convention which is negative sign for X why do we add regressors with plus in the R code? Either add more explanation to the comment or take it out

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd:1662]: There are some discrepancies between R and Python estimated values for beta. I think the issue is you forgot to change field_of_study into a factor in R. It seems like to be a str in python as well. I am guessing the mismatch between the type of variable in R and Pytohn is causing the trouble. You want to make sure they are both factors with different levels. You can see the estimates are the same for GPA and GRE_Score which suggests both functions fit the same mode - it seems like this is only a variable type issue. Can you check this?

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: In the fitted formula add the subscript for sample size i. Also I think you may want to explain more on the beta field transposed vector you defined. Let's try to explain it in a simple language accompanied with your math formula. The idea is to explain what is it doing? It is a vector with estimated values from field-of-study. Writing the values in a vector helps.

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd:1687]: You can refer to this formula by cross referencing it instead of writing it again

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Add a short explanation that later in the section you will explain how to run Brant test. You can use cross reference to that section once it is done

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: I like the idea of counting the number of parameters and making the case that this will be the cost we pay if we fit non-proportional mode. Let's add an example to show how these (J-1) \times (K+1) numbers come from. You can use the case study variables to list the number of parameters. Maybe with couple of bullet points?

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: I dont think we need this paragraph about matrix B. I think it is beyond the scope of this book. Maybe you can replace it with a short paragraph that says the optimization method is the same as before and make a cross reference to the previous tip box that you had

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Regarding this: "Setting `parallel = FALSE` " I do not see it in the documentation of vglm() function. Can you check again and add more details? Also explain what reverse = FALSE does here. A bit more information on how to use the function would be helpful

File: book/15-ordinal-logistic.qmd [hunk 6]
```diff
@@ -1458,7 +1459,7 @@ ordinal_fit_np <- vglm(
   data = train_df
 )
 
-summary(ordinal_fit_np)
+tidy_vgam(ordinal_fit_np) |> mutate_if(is.numeric, round, 3)
```
Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Let's keep the columns that we show between R and Pytohn the same. I like the Python output with 3 columns maybe you can do the same for R output + pvalue. Does python provide pvalue for test? If not I think you have to compute it manulaly for z-statistic.

File: book/15-ordinal-logistic.qmd
Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: For this "in `Python`, where no built-in non-proportional ordinal model exists," take a look at this Python package and see if it can helps?  https://pythonhosted.org/mord/

Also I think the change of variable type to factor also needed here. Maybe this explains the discrepencies among estimates

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Add this in the writing of this section too: "Unlike the usual R-squared in OLS regression, McFadden’s pseudo-R-squared should not be interpreted as the proportion of variance explained. Instead, it measures improvement in log-likelihood relative to the null model. Its values are typically much smaller than regular R0squared values, and values around 0.2–0.4 are often considered indicative of a good/descent model fit in discrete-response models."  Would be nice to add a reference here too

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Keep the notation the same, you are using LR statistic here but the formula was defined as G^2

Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: Cross reference to chapter 14 here with at sign

File: book/15-ordinal-logistic.qmd [hunk 7]
```diff
@@ -76,3 +113,2861 @@ mindmap
...
+
+## Inference
+
+After we finish estimating the coefficients, we examine how uncertain they are and whether they differ meaningfully from zero. We get a variance estimate for each coefficient from the observed information matrix (the negative of the Hessian at the MLE solution). Taking the square roots of those diagonal variances gives standard errors. Each coefficient divided by its standard error gives an asymptotically normal **Wald $z$-statistic** for testing whether the true effect is zero. Using the same standard errors, we build confidence intervals: estimate $\pm$ (critical value $\times$ standard error). To make results more interpretable, we exponentiate a regression coefficient (or its confidence thresholds) to obtain a **proportional odds ratio**, which describes how the cumulative odds of being at or below any category threshold change for a one-unit increase in the regressor, holding other variables fixed.
```
Comment (pnickchi) [inline book/15-ordinal-logistic.qmd]: True for a continous regressor, let's add a short sentence for interpretation of categorical regressors.

File: book/15-ordinal-logistic.qmd [hunk 8]
```diff
@@ -76,3 +113,2861 @@ mindmap
...
+::: {.panel-tabset}
+
+### R Code
+
+```{.r}
+library(MASS)
+library(broom)
+
+# polr() does not return p-values directly. tidy() gives the coefficient
+# table (estimates, standard errors, and t-values), from which we compute
```
Comment (pnickchi) [inline book/15-ordinal-logistic.qmd:2377]: Edit: from which we compute the approximate p-values based on Normal approximation

File: book/15-ordinal-logistic.qmd [hunk 9]
```diff
@@ -76,3 +113,2861 @@ mindmap
...
+$$
+
+which follows a $\chi^2$ distribution with $J - 2$ degrees of freedom under $H_0$ (proportional odds holds for $x_{i,k}$). An omnibus version of the test aggregates across all $K$ regressors, giving a $\chi^2$ statistic with $(J-2) \times K$ degrees of freedom.
+
+In `R`, the `brant()` function from the `brant` package computes this test directly on a `polr` object. In `Python`, we implement it manually by fitting three binary logistic regression models and comparing their slopes.
+
+::: {.panel-tabset}
+
+### R Code
+
```
Comment (pnickchi) [inline book/15-ordinal-logistic.qmd:2538]: The statistic value for omnibus and each regressor are not the same. Can you check please? I wonder if this is related to the type of variable discussion we had before.

File: book/15-ordinal-logistic.qmd
Comment (karanbayns) [reply book/15-ordinal-logistic.qmd]: I checked mord and unfortunately it can only fit a proportional ordinal logistic regression model.

**Comment by pnickchi** [comment]
Hi @karanbayns 

I have sent you this over email but I thought to share it here as well for your reference. Thanks for your all your hard work on chapter 15. I read the whole chapter and it looks great! I did not have much comment to add except some minor points which you can see below. I will make a final polish once we merge into main but things look good to me.

- In section 15.2 Case Study instead if "We will explore a dataset containing some" write it as "We will explore a simulated dataset containing some"

- In Table 15.1 both GPA and GRE score are bounded positive.

- In 15.5 Data Modelling section when you define the cumulative probability. Instead of writing it as j runs over the values of 1, 2, ..., J-1=2 write it as runs over values of 1,2 since J = 3

- Some of the references did not render for me locally e.g. see ?sec-binary-logistic) but I think this will be resolved when we merge in main - it is possible in the current branch we don't have those chapter references yet. No need to worry I check this again once the merge is done.

- For Tip on the Newton–Raphson Method please add a reference. You can include the textbook for ordinal regression that I shared with you.

- Please add numbers to the major formulas that you have in the textbook such as link function, statistic for hypothesis tests, confidence intervals, Brant–Wald Test, etc.

That is all from my end for now. Please apply these and let me know if you have any questions. I will review your multinomial chapter hopefully during this weekend and let you know of my comments during our Monday meeting at 10:00 am.
